api/_routers/97-require-access-token-allow-guest.go

```diff
@@ -0,0 +1,55 @@
+package _routers
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/getsentry/sentry-go"
+	"github.com/sirupsen/logrus"
+	"github.com/t2bot/matrix-media-repo/api/_apimeta"
+	"github.com/t2bot/matrix-media-repo/api/_auth_cache"
+	"github.com/t2bot/matrix-media-repo/api/_responses"
+	"github.com/t2bot/matrix-media-repo/common"
+	"github.com/t2bot/matrix-media-repo/common/config"
+	"github.com/t2bot/matrix-media-repo/common/rcontext"
+	"github.com/t2bot/matrix-media-repo/matrix"
+	"github.com/t2bot/matrix-media-repo/util"
+)
+
+func RequireAccessTokenAllowGuest(generator GeneratorWithUserFn) GeneratorFn {
```

Comment:
instead of another access control function, adding `allowGuests bool` to `RequireAccessToken` would be fine. This means updating all the other routes, but explicit is always good.

New signature would be `RequireAccessToken(allowGuests bool, generator GeneratorWithUserFn) GeneratorFn`

Reply:
5e064b9